scrypto-unit/src/test_runner.rs

```diff
@@ -103,53 +104,81 @@ impl Compile {
     }
 }
 
-pub struct TestRunner {
-    scrypto_interpreter: ScryptoInterpreter<DefaultWasmEngine>,
-    staged_substate_store_manager: StagedSubstateStoreManager<TypedInMemorySubstateStore>,
-    intent_hash_manager: TestIntentHashManager,
-    next_private_key: u64,
-    next_transaction_nonce: u64,
+pub struct TestRunnerBuilder {
```

Comment:
**Note from author:**

I introduced a builder (instead of combinatorial number of factory methods).
This is the only reason so many test files are touched by this PR.

Reply:
Looks nice! 👍